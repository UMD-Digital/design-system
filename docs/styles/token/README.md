[**UMD Styles Library**](../README.md)

***

[UMD Styles Library](../README.md) / token

# token

## Namespaces

| Namespace | Description |
| ------ | ------ |
| [font](namespaces/font/README.md) | Typography and font tokens. |
| [media](namespaces/media/README.md) | Media query breakpoints and utilities. |

## Variables

| Variable | Description |
| ------ | ------ |
| [color](variables/color.md) | Default color tokens for the design system. |
| [spacing](variables/spacing.md) | Default spacing tokens with semantic naming. |
