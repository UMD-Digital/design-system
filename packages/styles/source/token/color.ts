/**
 * @module token/color
 * Provides color tokens for the design system.
 */

export type ColorScale = typeof colorScale;
export type BaseColors = typeof baseColors;

/**
 * Color scale with various shades for different colors.
 * @returns {ColorScale} Color scale object with various color categories and shades.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.colorScale.gray[500]
 * ```
 * @since 1.1.0
 */
export const colorScale = {
  gray: {
    50: '#FAFAFA',
    100: '#F1F1F1',
    200: '#E6E6E6',
    400: '#7F7F7F',
    500: '#757575',
    700: '#454545',
    900: '#242424',
  },
  red: {
    500: '#E21833',
    900: '#A90007',
  },
  blue: {
    500: '#2F7EDA',
  },
  green: {
    500: '#008000',
  },
  gold: {
    500: '#FFD200',
  },
} as const;

/**
 * Base colors for black and white.
 * @returns {BaseColors} Base colors object with white and black.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.baseColors.white
 * ```
 * @since 1.1.0
 */
export const baseColors = {
  white: '#FFFFFF',
  black: '#000000',
} as const;

/**
 * Default color tokens for the design system.
 * @returns {object} Color tokens with semantic naming.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.color.red
 * ```
 * @since 1.1.0
 */
export default {
  red: colorScale.red[500],
  blue: colorScale.blue[500],
  green: colorScale.green[500],
  gold: colorScale.gold[500],
  white: baseColors.white,
  black: baseColors.black,
  redDark: colorScale.red[900],
  gray: {
    darker: colorScale.gray[900],
    dark: colorScale.gray[700],
    mediumAA: colorScale.gray[500],
    medium: colorScale.gray[400],
    light: colorScale.gray[200],
    lighter: colorScale.gray[100],
    lightest: colorScale.gray[50],
  },
};
