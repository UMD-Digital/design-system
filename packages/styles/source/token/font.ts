/**
 * @module token/font
 * Provides font-related design tokens including sizes, weights, and font families.
 */

/**
 * Font size tokens.
 * @type {Object<string, string>}
 * @property {string} min - 12px font size
 * @property {string} sm - 14px font size
 * @property {string} base - 16px font size (default)
 * @property {string} lg - 18px font size
 * @property {string} xl - 20px font size
 * @property {string} 2xl - 22px font size
 * @property {string} 3xl - 24px font size
 * @property {string} 4xl - 32px font size
 * @property {string} 5xl - 48px font size
 * @property {string} 6xl - 56px font size
 * @property {string} 7xl - 64px font size
 * @property {string} 8xl - 72px font size
 * @property {string} 9xl - 80px font size
 * @property {string} 10xl - 96px font size
 * @property {string} max - 120px font size
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.font.size.base // '16px'
 * ```
 * @since 1.8.0
 */
export const size = {
  min: '12px',
  sm: '14px',
  base: '16px',
  lg: '18px',
  xl: '20px',
  '2xl': '22px',
  '3xl': '24px',
  '4xl': '32px',
  '5xl': '48px',
  '6xl': '56px',
  '7xl': '64px',
  '8xl': '72px',
  '9xl': '80px',
  '10xl': '96px',
  max: '120px',
};

/**
 * Font weight tokens.
 * @type {Object<string, string>}
 * @property {string} thin - 100 weight
 * @property {string} extraLight - 200 weight
 * @property {string} light - 300 weight
 * @property {string} normal - 400 weight (default)
 * @property {string} medium - 500 weight
 * @property {string} semiBold - 600 weight
 * @property {string} bold - 700 weight
 * @property {string} extraBold - 800 weight
 * @property {string} black - 900 weight
 * @property {string} extraBlack - 950 weight
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.font.weight.bold // '700'
 * ```
 * @since 1.8.0
 */
export const weight = {
  thin: '100',
  extraLight: '200',
  light: '300',
  normal: '400',
  medium: '500',
  semiBold: '600',
  bold: '700',
  extraBold: '800',
  black: '900',
  extraBlack: '950',
};

/**
 * Font family tokens.
 * @type {Object<string, string>}
 * @property {string} campaign - Barlow Condensed font stack for campaign/marketing text
 * @property {string} sans - Interstate font stack for primary sans-serif text
 * @property {string} serif - Crimson Pro font stack for serif text
 * @property {string} mono - Monospace font stack for code/technical text
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.font.family.sans // "'Interstate', Helvetica, Arial, Verdana, sans-serif"
 * ```
 * @since 1.8.0
 */
export const family = {
  campaign: "'Barlow Condensed', Arial Narrow, sans-serif",
  sans: "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  serif: "'Crimson Pro', Georgia, serif",
  mono: 'monospace',
};
