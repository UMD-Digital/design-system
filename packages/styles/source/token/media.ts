/**
 * @module token/media
 * Provides responsive design tokens for breakpoints and media queries.
 */

/**
 * Numeric breakpoint values in pixels.
 * @type {Object<string, {min: number, max?: number}>}
 * @property {Object} small - Small screen breakpoint (320px-479px)
 * @property {Object} medium - Medium screen breakpoint (480px-649px)
 * @property {Object} large - Large screen breakpoint (650px-767px) 
 * @property {Object} tablet - Tablet breakpoint (768px-1023px)
 * @property {Object} desktop - Desktop breakpoint (1024px-1199px)
 * @property {Object} highDef - High-definition screen breakpoint (1200px-1499px)
 * @property {Object} maximum - Maximum screen size breakpoint (1500px+)
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.media.breakpointValues.desktop.min // 1024
 * ```
 * @since 1.8.0
 */
export const breakpointValues = {
  small: { min: 320, max: 479 },
  medium: { min: 480, max: 649 },
  large: { min: 650, max: 767 },
  tablet: { min: 768, max: 1023 },
  desktop: { min: 1024, max: 1199 },
  highDef: { min: 1200, max: 1499 },
  maximum: { min: 1500 },
};

/**
 * Breakpoint values with 'px' units for CSS usage.
 * @type {Object<string, {min: string, max?: string}>}
 * @property {Object} small - Small screen breakpoint strings ("320px"-"479px")
 * @property {Object} medium - Medium screen breakpoint strings ("480px"-"649px")
 * @property {Object} large - Large screen breakpoint strings ("650px"-"767px")
 * @property {Object} tablet - Tablet breakpoint strings ("768px"-"1023px")
 * @property {Object} desktop - Desktop breakpoint strings ("1024px"-"1199px")
 * @property {Object} highDef - High-definition screen breakpoint strings ("1200px"-"1499px")
 * @property {Object} maximum - Maximum screen size breakpoint string ("1500px"+)
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.media.breakpoints.desktop.min // "1024px"
 * ```
 * @since 1.8.0
 */
export const breakpoints = {
  small: {
    min: `${breakpointValues.small.min}px`,
    max: `${breakpointValues.small.max}px`,
  },
  medium: {
    min: `${breakpointValues.medium.min}px`,
    max: `${breakpointValues.medium.max}px`,
  },
  large: {
    min: `${breakpointValues.large.min}px`,
    max: `${breakpointValues.large.max}px`,
  },
  tablet: {
    min: `${breakpointValues.tablet.min}px`,
    max: `${breakpointValues.tablet.max}px`,
  },
  desktop: {
    min: `${breakpointValues.desktop.min}px`,
    max: `${breakpointValues.desktop.max}px`,
  },
  highDef: {
    min: `${breakpointValues.highDef.min}px`,
    max: `${breakpointValues.highDef.max}px`,
  },
  maximum: { min: `${breakpointValues.maximum.min}px` },
};

/**
 * Complete media query strings ready for use in CSS.
 * @type {Object<string, {min: string, max?: string}>}
 * @property {Object} small - Small screen media queries
 * @property {Object} medium - Medium screen media queries
 * @property {Object} large - Large screen media queries
 * @property {Object} tablet - Tablet media queries
 * @property {Object} desktop - Desktop media queries
 * @property {Object} highDef - High-definition screen media queries
 * @property {Object} maximum - Maximum screen size media query
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * // Use in JSS/CSS-in-JS
 * const styles = {
 *   [`@media (${Styles.token.media.queries.desktop.min})`]: {
 *     // Desktop-specific styles
 *   }
 * };
 * ```
 * @since 1.8.0
 */
export const queries = {
  small: {
    min: `min-width: ${breakpoints.small.min}`,
    max: `max-width: ${breakpoints.small.max}`,
  },
  medium: {
    min: `min-width: ${breakpoints.medium.min}`,
    max: `max-width: ${breakpoints.medium.max}`,
  },
  large: {
    min: `min-width: ${breakpoints.large.min}`,
    max: `max-width: ${breakpoints.large.max}`,
  },
  tablet: {
    min: `min-width: ${breakpoints.tablet.min}`,
    max: `max-width: ${breakpoints.tablet.max}`,
  },
  desktop: {
    min: `min-width: ${breakpoints.desktop.min}`,
    max: `max-width: ${breakpoints.desktop.max}`,
  },
  highDef: {
    min: `min-width: ${breakpoints.highDef.min}`,
    max: `max-width: ${breakpoints.highDef.max}`,
  },
  maximum: { min: `min-width: ${breakpoints.maximum.min}` },
};
