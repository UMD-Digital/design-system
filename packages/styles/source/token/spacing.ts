/**
 * @module token/spacing
 * Provides spacing tokens for the design system.
 */

/**
 * Spacing scale with values in pixels and their corresponding rem equivalents.
 * @returns {object} Spacing scale with numeric keys and pixel values.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.spacingScale[4] // '16px'
 * ```
 * @since 1.1.0
 */
export const spacingScale = {
  0: '0',
  1: '4px', // 0.25rem
  2: '8px', // 0.5rem  (min)
  3: '12px', // 0.75rem (xs)
  4: '16px', // 1rem    (sm)
  6: '24px', // 1.5rem  (md)
  8: '32px', // 2rem    (lg)
  10: '40px', // 2.5rem  (xl)
  12: '48px', // 3rem    (2xl)
  14: '56px', // 3.5rem  (3xl)
  16: '64px', // 4rem    (4xl)
  18: '72px', // 4.5rem  (5xl)
  20: '80px', // 5rem    (6xl)
  24: '96px', // 6rem    (7xl)
  26: '104px', // 6.5rem  (8xl)
  30: '120px', // 7.5rem  (max)
} as const;

/**
 * Maximum width constraints for layout containers.
 * @returns {object} Maximum width values with semantic naming.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.maxWidth.normal // '1384px'
 * ```
 * @since 1.1.0
 */
export const maxWidth = {
  max: '1728px',
  large: '1552px',
  normal: '1384px',
  small: '1200px',
  smallest: '992px',
};

/**
 * Default spacing tokens with semantic naming.
 * @returns {object} Spacing tokens with semantic names and corresponding pixel values.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.spacing.md // '24px'
 * ```
 * @since 1.1.0
 */
export default {
  min: '8px',
  xs: '12px',
  sm: '16px',
  md: '24px',
  lg: '32px',
  xl: '40px',
  '2xl': '48px',
  '3xl': '56px',
  '4xl': '64px',
  '5xl': '72px',
  '6xl': '80px',
  '7xl': '96px',
  '8xl': '104px',
  max: '120px',
  maxWidth: {
    ...maxWidth,
  },
};
