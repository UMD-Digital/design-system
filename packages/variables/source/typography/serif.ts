import { FontFamily, FontSize, FontWeight } from '../tokens/fonts';
import { Queries } from '../tokens/breakpoints';

const breakpointLarge = Queries.large.min;
const breakpointDesktop = Queries.desktop.min;

const FontBase = {
  fontFamily: FontFamily['serif'],
};

const SizeLarger = {
  fontSize: FontSize['4xl'],
  lineHeight: `1.18em`,
};
const SizeLarge = {
  fontSize: FontSize['3xl'],
  lineHeight: `1em`,
};
const SizeMedium = {
  fontSize: FontSize['xl'],
  lineHeight: `1.5em`,
};

export const SerifMaxium = {
  ...FontBase,
  ...SizeLarger,

  [`@media (${breakpointLarge})`]: {
    fontSize: `calc(${FontSize['4xl']} + 4vw)`,
  },

  [`@media (${breakpointDesktop})`]: {
    fontSize: FontSize['9xl'],
    fontWeight: FontWeight['light'],
    lineHeight: `1.025em`,
  },
};

export const SerifExtralarge = {
  ...FontBase,
  ...SizeLarger,

  [`@media (${breakpointLarge})`]: {
    fontSize: `calc(${FontSize['4xl']} + 2vw)`,
  },

  [`@media (${breakpointDesktop})`]: {
    fontSize: FontSize['6xl'],
    lineHeight: `1.07em`,
  },
};

export const SerifLarger = {
  ...FontBase,
  ...SizeLarge,

  [`@media (${breakpointLarge})`]: {
    fontSize: `calc(${FontSize['3xl']} + 0.66vw)`,
  },

  [`@media (${breakpointDesktop})`]: {
    ...SizeLarger,
  },
};

export const SerifLarge = {
  ...FontBase,
  ...SizeMedium,

  [`@media (${breakpointLarge})`]: {
    fontSize: `calc(${FontSize['xl']} + 0.33vw)`,
  },

  [`@media (${breakpointDesktop})`]: {
    ...SizeLarge,
  },
};

export const SerifMedium = {
  ...FontBase,
  ...SizeMedium,
};
