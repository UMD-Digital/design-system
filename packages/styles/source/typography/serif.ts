/**
 * @module typography/serif
 * Provides serif typography styles with various sizes and responsive scaling options.
 */

import { font, media } from '../token';
import { create } from '../utilities';

/**
 * Breakpoint variables for responsive typography.
 * @type {string}
 * @private
 */
const breakpointLarge = media.queries.large.min;
const breakpointDesktop = media.queries.desktop.min;

/**
 * Base serif typography properties.
 * @type {object}
 * @private
 */
const base = {
  fontFamily: font.family['serif'],
};

/**
 * Larger size serif typography constants.
 * @type {object}
 * @private
 */
const sizeLarger = {
  fontSize: font.size['4xl'],
  lineHeight: `1.18em`,
};

/**
 * Large size serif typography constants.
 * @type {object}
 * @private
 */
const sizeLarge = {
  fontSize: font.size['3xl'],
  lineHeight: `1em`,
};

/**
 * Medium size serif typography constants.
 * @type {object}
 * @private
 */
const sizeMedium = {
  fontSize: font.size['xl'],
  lineHeight: `1.5em`,
};

/**
 * Maximum size serif typography style with responsive scaling.
 * @type {object}
 * @property {string} fontFamily - Serif font family
 * @property {string} fontSize - Extra large size that scales dramatically at breakpoints
 * @property {string} lineHeight - Appropriate line height for heading text
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * // Use in object spread syntax
 * const myStyle = { ...Styles.typography.serif.maxium };
 * ```
 * @since 1.1.0
 */
export const maxium = {
  ...base,
  ...sizeLarger,

  [`@media (${breakpointLarge})`]: {
    fontSize: `calc(${font.size['4xl']} + 4vw)`,
  },

  [`@media (${breakpointDesktop})`]: {
    fontSize: font.size['9xl'],
    fontWeight: font.weight['light'],
    lineHeight: `1.025em`,
  },
};

/**
 * Extra large serif typography style with responsive scaling.
 * @type {object}
 * @property {string} fontFamily - Serif font family
 * @property {string} fontSize - Large size that scales up at breakpoints
 * @property {string} lineHeight - Appropriate line height for heading text
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * // Use in object spread syntax
 * const myStyle = { ...Styles.typography.serif.extralarge };
 * ```
 * @since 1.1.0
 */
export const extralarge = {
  ...base,
  ...sizeLarger,

  [`@media (${breakpointLarge})`]: {
    fontSize: `calc(${font.size['4xl']} + 2vw)`,
  },

  [`@media (${breakpointDesktop})`]: {
    fontSize: font.size['6xl'],
    lineHeight: `1.07em`,
  },
};

/**
 * Larger serif typography style with responsive scaling.
 * @type {object}
 * @property {string} fontFamily - Serif font family
 * @property {string} fontSize - Large size that scales up at breakpoints
 * @property {string} lineHeight - Appropriate line height for heading text
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * // Use in object spread syntax
 * const myStyle = { ...Styles.typography.serif.larger };
 * ```
 * @since 1.1.0
 */
export const larger = {
  ...base,
  ...sizeLarge,

  [`@media (${breakpointLarge})`]: {
    fontSize: `calc(${font.size['3xl']} + 0.66vw)`,
  },

  [`@media (${breakpointDesktop})`]: {
    ...sizeLarger,
  },
};

/**
 * Large serif typography style with responsive scaling.
 * @type {object}
 * @property {string} fontFamily - Serif font family
 * @property {string} fontSize - Medium size that scales up to large at breakpoints
 * @property {string} lineHeight - Appropriate line height for text
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * // Use in object spread syntax
 * const myStyle = { ...Styles.typography.serif.large };
 * ```
 * @since 1.1.0
 */
export const large = {
  ...base,
  ...sizeMedium,

  [`@media (${breakpointLarge})`]: {
    fontSize: `calc(${font.size['xl']} + 0.33vw)`,
  },

  [`@media (${breakpointDesktop})`]: {
    ...sizeLarge,
  },
};

/**
 * Medium serif typography style.
 * @type {object}
 * @property {string} fontFamily - Serif font family
 * @property {string} fontSize - Medium font size
 * @property {string} lineHeight - Comfortable line height for reading
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * // Use in object spread syntax
 * const myStyle = { ...Styles.typography.serif.medium };
 * ```
 * @since 1.1.0
 */
export const medium = {
  ...base,
  ...sizeMedium,
};

/**
 * Ready-to-use serif typography styles as JSS objects with class names.
 * @type {Object<string, JssObject>}
 * @property {JssObject} maximum - JSS object for maximum size text
 * @property {JssObject} extraLarge - JSS object for extra large size text
 * @property {JssObject} larger - JSS object for larger size text
 * @property {JssObject} large - JSS object for large size text
 * @property {JssObject} medium - JSS object for medium size text
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * // Use as a complete JSS object including className
 * Styles.typography.serif.fonts.large
 * ```
 * @since 1.1.0
 */
export const fonts = {
  maximum: create.jss.objectWithClassName({
    className: 'umd-serif-maximum',
    ...maxium,
  }),

  extraLarge: create.jss.objectWithClassName({
    className: 'umd-serif-extralarge',
    ...extralarge,
  }),

  larger: create.jss.objectWithClassName({
    className: 'umd-serif-larger',
    ...larger,
  }),

  large: create.jss.objectWithClassName({
    className: 'umd-serif-large',
    ...large,
  }),

  medium: create.jss.objectWithClassName({
    className: 'umd-serif-medium',
    ...medium,
  }),
};
