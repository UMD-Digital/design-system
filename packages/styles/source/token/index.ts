/**
 * @module token
 * Design tokens and system variables.
 */

import color from './color';
import spacing from './spacing';

/**
 * Media query breakpoints and utilities.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.media
 * ```
 * @since 1.1.0
 */
export * as media from './media';

/**
 * Typography and font tokens.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.font
 * ```
 * @since 1.1.0
 */
export * as font from './font';

/**
 * Color system tokens.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.color
 * ```
 * @since 1.1.0
 */
export { color };

/**
 * Spacing system tokens.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.spacing
 * ```
 * @since 1.1.0
 */
export { spacing };
