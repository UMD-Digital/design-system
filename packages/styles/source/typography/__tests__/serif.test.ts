import * as serif from '../serif';
import { font, media } from '../../token';

describe('typography/serif', () => {
  it('should match snapshot', () => {
    expect(serif).toMatchSnapshot();
  });

  describe('size objects', () => {
    it('should have correctly defined serif sizes', () => {
      // Test individual size exports
      expect(serif.maxium).toMatchSnapshot();
      expect(serif.extralarge).toMatchSnapshot();
      expect(serif.larger).toMatchSnapshot();
      expect(serif.large).toMatchSnapshot();
      expect(serif.medium).toMatchSnapshot();
    });

    it('should use the serif font family for all sizes', () => {
      // All sizes should use serif font family
      [
        serif.maxium,
        serif.extralarge,
        serif.larger,
        serif.large,
        serif.medium,
      ].forEach((size) => {
        // Using the base object inside each size
        expect(size.fontFamily).toBe(font.family.serif);
      });
    });
  });

  describe('fonts objects', () => {
    it('should have correctly named font objects with classNames', () => {
      expect(serif.fonts.maximum.className).toBe('umd-serif-maximum');
      expect(serif.fonts.extraLarge.className).toBe('umd-serif-extralarge');
      expect(serif.fonts.larger.className).toBe('umd-serif-larger');
      expect(serif.fonts.large.className).toBe('umd-serif-large');
      expect(serif.fonts.medium.className).toBe('umd-serif-medium');
    });

    it('should match snapshot for all fonts objects', () => {
      expect(serif.fonts).toMatchSnapshot();
    });
  });
});
