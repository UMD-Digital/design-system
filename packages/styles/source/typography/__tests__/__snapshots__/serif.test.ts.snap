// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typography/serif fonts objects should match snapshot for all fonts objects 1`] = `
{
  "extraLarge": {
    "@media (min-width: 1024px)": {
      "fontSize": "56px",
      "lineHeight": "1.07em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(32px + 2vw)",
    },
    "className": "umd-serif-extralarge",
    "fontFamily": "'Crimson Pro', Georgia, serif",
    "fontSize": "32px",
    "lineHeight": "1.18em",
  },
  "large": {
    "@media (min-width: 1024px)": {
      "fontSize": "24px",
      "lineHeight": "1em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(20px + 0.33vw)",
    },
    "className": "umd-serif-large",
    "fontFamily": "'Crimson Pro', Georgia, serif",
    "fontSize": "20px",
    "lineHeight": "1.5em",
  },
  "larger": {
    "@media (min-width: 1024px)": {
      "fontSize": "32px",
      "lineHeight": "1.18em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(24px + 0.66vw)",
    },
    "className": "umd-serif-larger",
    "fontFamily": "'Crimson Pro', Georgia, serif",
    "fontSize": "24px",
    "lineHeight": "1em",
  },
  "maximum": {
    "@media (min-width: 1024px)": {
      "fontSize": "80px",
      "fontWeight": "300",
      "lineHeight": "1.025em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(32px + 4vw)",
    },
    "className": "umd-serif-maximum",
    "fontFamily": "'Crimson Pro', Georgia, serif",
    "fontSize": "32px",
    "lineHeight": "1.18em",
  },
  "medium": {
    "className": "umd-serif-medium",
    "fontFamily": "'Crimson Pro', Georgia, serif",
    "fontSize": "20px",
    "lineHeight": "1.5em",
  },
}
`;

exports[`typography/serif should match snapshot 1`] = `
{
  "extralarge": {
    "@media (min-width: 1024px)": {
      "fontSize": "56px",
      "lineHeight": "1.07em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(32px + 2vw)",
    },
    "fontFamily": "'Crimson Pro', Georgia, serif",
    "fontSize": "32px",
    "lineHeight": "1.18em",
  },
  "fonts": {
    "extraLarge": {
      "@media (min-width: 1024px)": {
        "fontSize": "56px",
        "lineHeight": "1.07em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(32px + 2vw)",
      },
      "className": "umd-serif-extralarge",
      "fontFamily": "'Crimson Pro', Georgia, serif",
      "fontSize": "32px",
      "lineHeight": "1.18em",
    },
    "large": {
      "@media (min-width: 1024px)": {
        "fontSize": "24px",
        "lineHeight": "1em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(20px + 0.33vw)",
      },
      "className": "umd-serif-large",
      "fontFamily": "'Crimson Pro', Georgia, serif",
      "fontSize": "20px",
      "lineHeight": "1.5em",
    },
    "larger": {
      "@media (min-width: 1024px)": {
        "fontSize": "32px",
        "lineHeight": "1.18em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(24px + 0.66vw)",
      },
      "className": "umd-serif-larger",
      "fontFamily": "'Crimson Pro', Georgia, serif",
      "fontSize": "24px",
      "lineHeight": "1em",
    },
    "maximum": {
      "@media (min-width: 1024px)": {
        "fontSize": "80px",
        "fontWeight": "300",
        "lineHeight": "1.025em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(32px + 4vw)",
      },
      "className": "umd-serif-maximum",
      "fontFamily": "'Crimson Pro', Georgia, serif",
      "fontSize": "32px",
      "lineHeight": "1.18em",
    },
    "medium": {
      "className": "umd-serif-medium",
      "fontFamily": "'Crimson Pro', Georgia, serif",
      "fontSize": "20px",
      "lineHeight": "1.5em",
    },
  },
  "large": {
    "@media (min-width: 1024px)": {
      "fontSize": "24px",
      "lineHeight": "1em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(20px + 0.33vw)",
    },
    "fontFamily": "'Crimson Pro', Georgia, serif",
    "fontSize": "20px",
    "lineHeight": "1.5em",
  },
  "larger": {
    "@media (min-width: 1024px)": {
      "fontSize": "32px",
      "lineHeight": "1.18em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(24px + 0.66vw)",
    },
    "fontFamily": "'Crimson Pro', Georgia, serif",
    "fontSize": "24px",
    "lineHeight": "1em",
  },
  "maxium": {
    "@media (min-width: 1024px)": {
      "fontSize": "80px",
      "fontWeight": "300",
      "lineHeight": "1.025em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(32px + 4vw)",
    },
    "fontFamily": "'Crimson Pro', Georgia, serif",
    "fontSize": "32px",
    "lineHeight": "1.18em",
  },
  "medium": {
    "fontFamily": "'Crimson Pro', Georgia, serif",
    "fontSize": "20px",
    "lineHeight": "1.5em",
  },
}
`;

exports[`typography/serif size objects should have correctly defined serif sizes 1`] = `
{
  "@media (min-width: 1024px)": {
    "fontSize": "80px",
    "fontWeight": "300",
    "lineHeight": "1.025em",
  },
  "@media (min-width: 650px)": {
    "fontSize": "calc(32px + 4vw)",
  },
  "fontFamily": "'Crimson Pro', Georgia, serif",
  "fontSize": "32px",
  "lineHeight": "1.18em",
}
`;

exports[`typography/serif size objects should have correctly defined serif sizes 2`] = `
{
  "@media (min-width: 1024px)": {
    "fontSize": "56px",
    "lineHeight": "1.07em",
  },
  "@media (min-width: 650px)": {
    "fontSize": "calc(32px + 2vw)",
  },
  "fontFamily": "'Crimson Pro', Georgia, serif",
  "fontSize": "32px",
  "lineHeight": "1.18em",
}
`;

exports[`typography/serif size objects should have correctly defined serif sizes 3`] = `
{
  "@media (min-width: 1024px)": {
    "fontSize": "32px",
    "lineHeight": "1.18em",
  },
  "@media (min-width: 650px)": {
    "fontSize": "calc(24px + 0.66vw)",
  },
  "fontFamily": "'Crimson Pro', Georgia, serif",
  "fontSize": "24px",
  "lineHeight": "1em",
}
`;

exports[`typography/serif size objects should have correctly defined serif sizes 4`] = `
{
  "@media (min-width: 1024px)": {
    "fontSize": "24px",
    "lineHeight": "1em",
  },
  "@media (min-width: 650px)": {
    "fontSize": "calc(20px + 0.33vw)",
  },
  "fontFamily": "'Crimson Pro', Georgia, serif",
  "fontSize": "20px",
  "lineHeight": "1.5em",
}
`;

exports[`typography/serif size objects should have correctly defined serif sizes 5`] = `
{
  "fontFamily": "'Crimson Pro', Georgia, serif",
  "fontSize": "20px",
  "lineHeight": "1.5em",
}
`;
